Jenkinsfile

```diff
@@ -22,65 +23,86 @@ try {
+                  "-v /dev/bus/usb:/dev/bus/usb " +
+                  "-v ${env.HOME}/gradle-cache:/tmp/.gradle " +
+                  "-v ${env.HOME}/.android:/tmp/.android " +
+                  "-v ${env.HOME}/ccache:/tmp/.ccache " +
+                  "--network container:ros") {
+            stage('JVM tests') {
+              try {
+                withCredentials([[$class: 'FileBinding', credentialsId: 'c0cc8f9e-c3f1-4e22-b22f-6568392e26ae', variable: 'S3CFG']]) {
+                  sh "chmod +x gradlew && ./gradlew assemble check javadoc -Ps3cfg=${env.S3CFG}"
+                }
+              } finally {
+                storeJunitResults 'realm/realm-annotations-processor/build/test-results/test/TEST-*.xml'
+                storeJunitResults 'examples/unitTestExample/build/test-results/**/TEST-*.xml'
+                step([$class: 'LintPublisher'])
+              }
             }
-          } finally {
-            storeJunitResults 'realm/realm-annotations-processor/build/test-results/test/TEST-*.xml'
-            storeJunitResults 'examples/unitTestExample/build/test-results/**/TEST-*.xml'
-            step([$class: 'LintPublisher'])
-          }
-        }
-
-        stage('Static code analysis') {
-          try {
-            gradle('realm', 'findbugs pmd checkstyle')
-          } finally {
-            publishHTML(target: [allowMissing: false, alwaysLinkToLastBuild: false, keepAll: true, reportDir: 'realm/realm-library/build/findbugs', reportFiles: 'findbugs-output.html', reportName: 'Findbugs issues'])
-            publishHTML(target: [allowMissing: false, alwaysLinkToLastBuild: false, keepAll: true, reportDir: 'realm/realm-library/build/reports/pmd', reportFiles: 'pmd.html', reportName: 'PMD Issues'])
-            step([$class: 'CheckStylePublisher',
-            canComputeNew: false,
-            defaultEncoding: '',
-            healthy: '',
-            pattern: 'realm/realm-library/build/reports/checkstyle/checkstyle.xml',
-            unHealthy: ''
-            ])
-          }
-        }
-
-        stage('Run instrumented tests') {
-          boolean archiveLog = true
-          String backgroundPid
-          try {
-            backgroundPid = startLogCatCollector()
-            gradle('realm', 'connectedUnitTests')
-            archiveLog = false;
-          } finally {
-            stopLogCatCollector(backgroundPid, archiveLog)
-            storeJunitResults 'realm/realm-library/build/outputs/androidTest-results/connected/**/TEST-*.xml'
-          }
-        }
 
-        // TODO: add support for running monkey on the example apps
+            stage('Static code analysis') {
+              try {
+                gradle('realm', 'findbugs pmd checkstyle')
+              } finally {
+                publishHTML(target: [allowMissing: false, alwaysLinkToLastBuild: false, keepAll: true, reportDir: 'realm/realm-library/build/findbugs', reportFiles: 'findbugs-output.html', reportName: 'Findbugs issues'])
```

Comment:
No need. the Lint plugin takes care of this already: https://ci.realm.io/job/realm/job/realm-java/job/master/166/androidLintResult/